client/config/config.go

```diff
@@ -0,0 +1,183 @@
+package config
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	tmcli "github.com/tendermint/tendermint/libs/cli"
+)
+
+// Default constants
+const (
+	chainID        = ""
+	keyringBackend = "os"
+	output         = "text"
+	node           = "tcp://localhost:26657"
+	broadcastMode  = "sync"
+	trace          = false
+)
+
+var ErrWrongNumberOfArgs = fmt.Errorf("wrong number of arguments")
+
+type ClientConfig struct {
+	ChainID        string `mapstructure:"chain-id" json:"chain-id"`
+	KeyringBackend string `mapstructure:"keyring-backend" json:"keyring-backend"`
+	Output         string `mapstructure:"output" json:"output"`
+	Node           string `mapstructure:"node" json:"node"`
+	BroadcastMode  string `mapstructure:"broadcast-mode" json:"broadcast-mode"`
+	Trace          bool   `mapstructure:"trace" json:"trace"`
+}
+
+// TODO Validate values in setters
+func (c *ClientConfig) SetChainID(chainID string) {
+	c.ChainID = chainID
+}
+
+func (c *ClientConfig) SetKeyringBackend(keyringBackend string) {
+	c.KeyringBackend = keyringBackend
+}
+
+func (c *ClientConfig) SetOutput(output string) {
+	c.Output = output
+}
+
+func (c *ClientConfig) SetNode(node string) {
+	c.Node = node
+}
+
+func (c *ClientConfig) SetBroadcastMode(broadcastMode string) {
+	c.BroadcastMode = broadcastMode
+}
+
+func (c *ClientConfig) SetTrace(trace string) error {
+	boolVal, err := strconv.ParseBool(trace)
+	if err != nil {
+		return err
+	}
+	c.Trace = boolVal
+	return nil
+}
+
+func DefaultClientConfig() *ClientConfig {
+	return &ClientConfig{chainID, keyringBackend, output, node, broadcastMode, trace}
+}
+
+// Cmd returns a CLI command to interactively create an application CLI
+// config file.
+func Cmd(defaultCLIHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config <key> [value]",
+		Short: "Create or query an application CLI configuration file",
+		RunE:  runConfigCmd,
+		Args:  cobra.RangeArgs(0, 2),
+	}
+
+	cmd.Flags().String(flags.FlagHome, defaultCLIHome,
+		"set client's home directory for configuration")
+	return cmd
+}
+
+func runConfigCmd(cmd *cobra.Command, args []string) error {
+
+	v := viper.New()
+
+	cfgPath, err := ensureCfgPath(v.GetString(flags.FlagHome))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to make config path: %v\n", err)
+		return err
+	}
+
+	cliConfig, err := getClientConfig(cfgPath, v)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to get client config: %v\n", err)
+		return err
```

Comment:
In this case, thanks to cobra you should just do this (no need to print):

```suggestion
		return fmt.Errorf("couldn't get client config: %v", err)
```

Reply:
Done!